loopy/transform/iname.py

```diff
@@ -293,7 +293,14 @@ def _split_iname_backend(kernel, iname_to_split,
                 new_prio = new_prio + (prio_iname,)
         new_priorities.append(new_prio)
 
+    new_inames = kernel.inames.copy()
+    del new_inames[iname_to_split]
```

Comment:
What if `within` leaves out a few users of `iname_to_split`, then we might have to leave the call to `remove_unused_inames` in. I am working on making `remove_used_inames` less expensive, maybe that could help us  afford the call to `remove_unused_inames` via some early exit strategies.

Reply:
I don't understand how `within` is supposed to work. If the domains are splitted, then `iname_to_split` doesn't really exist anymore in domains and then loopy doesn't think it's an iname anymore right?

Reply:
> If the domains are splitted, then iname_to_split doesn't really exist anymore in domains and then loopy doesn't think it's an iname anymore right?

Not quite true, the `within` argument is generally passed to describe the granularity of a transformation. See [this example](https://github.com/inducer/loopy/blob/15064e22eb872fc3a9a049ce0821ea2ad8f57809/test/test_transform.py#L534-L548).

Reply:
Makes sense